bin/python/combine_reports_from_date_range.py

```diff
@@ -46,34 +48,48 @@ def _render_results(results):
 
     return "\n%s\n" * 3 % (tests_table.render(), categories_table.render(), missing_table.render())
 
-def _get_local_reports(device, reports_dir, start, end):
+
+def _iso_to_fname(timestamp):
+    return timestamp.isoformat().replace(':', '') if timestamp else None
+
+
+def _get_local_reports(device, reports_dir, start, end, count):
     LOGGER.info('Looking for reports locally')
     report_re = re.compile(r'^report_%s_\d{4}-\d{2}-\d{2}T\d{6}\+\d{4}.*\.json$' % device)
     ts_re = re.compile(r'\d{4}-\d{2}-\d{2}T\d{6}\+\d{4}')
     json_files = [f for f in os.listdir(reports_dir) if report_re.match(f)]
+    json_files.sort()
+    if count and len(json_files) > count:
+        json_files = json_files[len(json_files) - count:]
     for json_file in json_files:
         timestamp = ts_re.search(json_file).group(0)
-        start_str = start.isoformat().replace(':', '') if start else None
-        end_str = end.isoformat().replace(':', '') if end else None
+        start_str = _iso_to_fname(start)
+        end_str = _iso_to_fname(end)
         if (start_str and timestamp < start_str) or (end_str and timestamp > end_str):
             LOGGER.info('Skipping file %s' % json_file)
             continue
         LOGGER.info('Processing file %s' % json_file)
         with open(os.path.join(reports_dir, json_file), 'r') as json_file_handler:
             yield json.loads(json_file_handler.read())
 
-def main(device, start=None, end=None, gcp=None, reports_dir=os.path.join('inst', 'reports')):
+
+def main(device, start=None, end=None, gcp=None, reports_dir=DEFAULT_REPORTS_DIR, count=0):
+    # pylint: disable=too-many-arguments
     # pylint: disable=too-many-locals
     """Main script function"""
-    aggregate = {'tests': {}, 'categories': {}, 'missing': {}}
+
     device = device.replace(':', '').lower()
     if gcp:
         device_full = ":".join([device[i:i + 2] for i in range(0, len(device), 2)])
-        json_reports = gcp.get_reports_from_date_range(device_full, start=start, end=end)
+        json_reports = gcp.get_reports_from_date_range(device_full, start=start, end=end,
+                                                       count=count)
     else:
-        json_reports = _get_local_reports(device, reports_dir, start, end)
+        json_reports = _get_local_reports(device, reports_dir, start, end, count)
 
+    aggregate = {'tests': {}, 'categories': {}, 'missing': {}}
+    report_count = 0
     for json_report in json_reports:
+        report_count += 1
```

Comment:
report_count is not really necessary since report_count is len(json_reports)

Reply:
Kinda -- json_reports is actual an iterator so... doesn't like to use it twice.  But, I fixed that with a list() call instead of explicit count. Maybe less efficient but more straightforward.

Reply:
I forgot I used a generator on purpose here